packaging/irods_prc_tests/test_rule_engine_plugin_metadata_guard_atomic.py

```diff
@@ -0,0 +1,126 @@
+from irods.access import iRODSAccess
+from irods.session import iRODSSession
+from irods.meta import iRODSMeta, AVUOperation
+
+import irods.exception
+
+import json
+import os
+import shutil
+import tempfile
+import unittest
+
+class test_metadata_guard(unittest.TestCase):
+
+    IRODS_CONFIG_FILE_PATH = '/etc/irods/server_config.json'
+    TEST_DATA_OBJECT_PATH = '/tempZone/home/alice/test1'
+
+    @classmethod
+    def backup_irods_config(cls):
+        cls.temp_path = os.path.join(tempfile.mkdtemp(), 'irods_config_backup')
+        shutil.copy2(cls.IRODS_CONFIG_FILE_PATH, cls.temp_path)
+
+    @classmethod
+    def restore_irods_config(cls):
+        shutil.copy2(cls.temp_path,cls.IRODS_CONFIG_FILE_PATH)
+
+    @classmethod
+    def setUpClass(cls):
+        with iRODSSession(host='localhost', port=1247, user='rods', password='rods', zone='tempZone') as local_admin:
+            # create unprivileged user
+            local_admin.users.create_with_password('alice', 'test')
+
+            # unprivileged user creates test file
+            with iRODSSession(host='localhost', port=1247, user='alice', password='test', zone='tempZone') as unprivileged_user:
+                unprivileged_user.data_objects.create(cls.TEST_DATA_OBJECT_PATH)
+                unprivileged_user.acls.set( iRODSAccess('own', cls.TEST_DATA_OBJECT_PATH, 'rods', 'tempZone'))
+
+            coll = local_admin.collections.get("/tempZone")
+            # add metadata_guard config AVU to collection
+            METADATA_GUARD_CONFIG = {
+                'admin_only': True,
+                'prefixes': ['irods::']
+            }
+            if 'irods::metadata_guard' not in coll.metadata:
+                coll.metadata.add('irods::metadata_guard', json.dumps(METADATA_GUARD_CONFIG))
+
+            # create backup of irods config file
+            cls.backup_irods_config()
+            # insert metadata guard rule engine plugin to irods config file
+            with open(cls.IRODS_CONFIG_FILE_PATH, 'r+') as config_file:
+                METADATA_GUARD_PLUGIN_BLOCK = {
+                    'instance_name': 'irods_rule_engine_plugin-metadata_guard-instance',
+                    'plugin_name': 'irods_rule_engine_plugin-metadata_guard',
+                    'plugin_specific_configuration': {}
+                }
+                config_json = json.load(config_file)
+                config_json['plugin_configuration']['rule_engines'].insert(0, METADATA_GUARD_PLUGIN_BLOCK)
+            with open(cls.IRODS_CONFIG_FILE_PATH, 'wt') as config_file:
+                json.dump(config_json, config_file)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.restore_irods_config()
+
+    def test_bad_config(self):
```

Comment:
Append issue number to test name.

The convention for that is `<test_name>__issue_12345`.